contrib/drf_introspection/serializers.py

```diff
@@ -50,7 +50,35 @@ def get_allowed_keys(self):
         return set(self.fields.keys()) | self._extra_fields
 
 
-class StrictSerializerMixin(IntrospectableSerializerMixin):
+class SelectFieldsSerializerMixin(object):
+    """
+    Allows to select fields to display with 'fields'.
+
+    This can improve performance when omitting fields which would require
+    additional SQL queries to display.
+    """
+    def __init__(self, *args, **kwargs):
+        super(SelectFieldsSerializerMixin, self).__init__(*args, **kwargs)
+
+        try:
+            request = self.context['request']
+        except KeyError:
+            return
+
+        fields = request.query_params.get('fields')
```

Comment:
This fields name seems duplicate with pdc/apps/component/test.py line 491.

Reply:
Seems that the mixin for selecting fields already exists ([pdc/apps/common/serializers.py:13](https://github.com/product-definition-center/product-definition-center/blob/e67264e2a18171f3a54fc6f8f709d87c8db67b6d/pdc/apps/common/serializers.py#L13))

@simozhan Any ideas why `fields` are not available for all endpoints?

Reply:
@hluk I don't remember why 'fields' are not applied to all the endpoints. Maybe some apis do have this requirement when designing the API(maybe for performance issue), or maybe our original idea is just add this 'fields' API by API, sorry, can't recall it. Do you want to add this feature one place for all APIs? I don't have strong opinion about that, but if you want to do that, maybe we need to remove duplicate DynamicFieldsSerializerMixin.

Reply:
I don't see any issue with having `fields` and `exclude_fields` consistently in all APIs as with `ordering`.

I think I could make `DynamicFieldsSerializerMixin` base class of `StrictSerializerMixin` which is used for most serializers. And I'll add tests for this.

Reply:
@hluk Fine for me for making fields and exclude_fields consistently in all APIs